recorder.go

```diff
@@ -104,3 +105,72 @@ func (p *Printer) Record(loc *Location, op Operation, stats *Stats) error {
 	p.lastValue = time.Now()
 	return nil
 }
+
+// Full Recorder records all of the evaluations that occur during an optimization
```

Comment:
I understand your point about things being Ored and this not handling that properly. Do you have other suggestions for making it more general? It seems reasonable to me to provide some built-in way to store the history of the optimization for retrospection (in addition to the human-friendly printing we have now).